aemedge/blocks/channel-lookup/channel-lookup.js

```diff
@@ -0,0 +1,114 @@
+import { createTag } from '../../scripts/utils.js';
+
+const ZIPCODE_ENDPOINT = 'https://p-geo.movetv.com/geo';
+const DEFAULT_ZIPCODE = '90020';
+const ZIPCODE_KEY = 'user_zip';
+let gqlResponse = '';
+
+export async function getZipcode() {
+  let zipcode = localStorage.getItem(ZIPCODE_KEY);
+  if (!zipcode) {
+    const response = await fetch(ZIPCODE_ENDPOINT);
+    const data = await response.json();
+    zipcode = data?.zip_code || DEFAULT_ZIPCODE;
+    localStorage.setItem(ZIPCODE_KEY, zipcode);
+  }
+  return zipcode;
+}
+function showWarningMessage() {
+  return `
+    <div type="warning" class="warning channel-data">
+      <img src="/aemedge/icons/warning-icon.svg" id="icon" class="warning-icon"/>
+      <h3 class="warning-heading">
+      <span>
+      <p class="warning-msg">Local programming
+      is not available with Sling in your area. You can still watch the Olympics on USA on-demand.
+      </p>
+      </span>
+      </h3>
+    </div>`;
+}
+
+function channelName(packages) {
+  let channelDiv;
+  packages.forEach((pkg) => {
+    if (channelDiv !== undefined) {
+      channelDiv += `<picture><img class="channel-img"
+                      src="https://www.sling.com/content/dam/sling-tv/save-this/${pkg.name}.svg"></picture>`;
+    } else {
+      channelDiv = `<picture><img class="channel-img"
+                      src="https://www.sling.com/content/dam/sling-tv/save-this/${pkg.name}.svg"></picture>`;
+    }
+  });
+  return channelDiv;
+}
+async function getChannelsByZipCode(block) {
+  const zipCode = block.querySelector('#zipcode').value;
+  localStorage.setItem(ZIPCODE_KEY, zipCode);
+  const API_ENDPOINT = 'https://webapi.p.sling.com/catalog/packages/local_and_rsn';
+  try {
+    const res = await fetch(`${API_ENDPOINT}?zipcode=${zipCode}`);
+    const formContainer = block.querySelector('.channel-form-container');
+    if (res.ok) {
+      gqlResponse = await res.json();
+      if (gqlResponse.packages) {
+        if (block.querySelector('.channel-data')) {
+          document.querySelector('.channel-data').remove();
```

Comment:
suggest to reduce the scope of querySelector to "block" rather than "document". The reason is , for instance if there are two Channel Look up blocks on the same page , it will remove the first one and leave the rest

Reply:
done, please review